contrib/opencensus-ext-azure/opencensus/ext/azure/log_exporter/__init__.py

```diff
@@ -106,6 +106,20 @@ def log_record_to_envelope(self, record):
         raise NotImplementedError  # pragma: NO COVER
 
     def flush(self, timeout=None):
+        if self._queue.is_empty():
+            return
```

Comment:
It looks like Queue.flush() already checks for this condition

Reply:
It does. I added this check only to not `log.warning` about the pending messages in vain.